### DISCUSSION TITLE
there something similar to tmuxinator in zellij?

### DISCUSSION THREAD

**Discussion by sidneip** [zellij-org/zellij#2960] [General]
I am migrating from tmux and I am missing something tmuxinator to restore sessions and start sessions with preset configs.

I was seeing the layouts, is this similar tmuxinator ?

**Comment by Lockszmith-GH** [comment]
Currently Zellij has somehting called [Session Ressurection](https://zellij.dev/documentation/session-resurrection.html) which is on by default.

Layouts are a way to design predefined states that you can either have loaded by default or loaded by demand.

**Reply by logoff** [reply]
Thanks for the suggestion, @Lockszmith-GH. However, is there something like launching a session and executing a command on each tab, like tmuxinator offers?

**Reply by sidneip** [reply]
i did migrate to zellij and i am using ressurection and this is serving to me.
I did all sessions possible with my projects and do ctrl + o and w to start and change sessions.

But i think if i start all sessions projects will be bad because i have projects that i dont work every month i open 1 time at year and i would like to run this project this way ze start hour-lib-example and it run my template and ressurect session if it exist or create if dont exist.

**Reply by Lockszmith-GH** [reply]
Does #3203 answer your problem?

**Reply by sidneip** [reply]
Thank you

I believe that it is well to me. In fact I am already satisfied with currently way.

**Reply by Lockszmith-GH** [reply]
There is also the [welcome screen](https://github.com/zellij-org/zellij/pull/3112) with it's session selector.

I'm sorry I'm not more helpful, but my use of zellij is completley mundane - I use it as an adhoc, mulitplexer, no predefined layouts - probably because I tend to jump from subject to subject on my servers, and I'm not developing any single thing there.

Glad you're finding zellij useful, I know I have since I started using it a couple of years back.

**Comment by imsnif** [comment]
Check out layouts: https://zellij.dev/tutorials/layouts/
You can also pair them with session management: https://zellij.dev/tutorials/session-management/

**Comment by corvofeng** [comment]
Hello, everyone! 🎉

I've updated [kubemux](https://github.com/corvofeng/kubemux) to include support for Zellij.

Check it out in action:

[![asciicast](https://asciinema.org/a/693805.svg)](https://asciinema.org/a/693805)
This update aims to make it easier to reuse your existing tmuxinator configurations.

For the complete documentation, visit:
https://kubemux.corvo.fun/en/usage/zellij/

**Reply by Lockszmith-GH** [reply]
Very cool tool.
However, **not cool** hijacking an open, unrelated, discussion.

~~Please remove your comment here, and feel free to open a new discussion.~~
It would be better suited as a separate discussion, as it will have more eyes on it, and might benefit others. (and it's ok to remove it from here, even though it will remove my comments as well.)

**Reply by giyany** [reply]
Hi, 
The expectation to delete or remove comments only applies in code of conduct violations. 
Off-topic comments do not violate the zellij code of conduct, which you can find here:
https://github.com/zellij-org/zellij/blob/main/CODE_OF_CONDUCT.md

**Reply by Lockszmith-GH** [reply]
Thanks, @giyany . My intention was to make a recommendation, but it turned out too harsh - text is such a finicky medium. Sorry about that.
